src/testnet/helium/event_dispatcher.rs

```diff
@@ -0,0 +1,242 @@
+use std::collections::hash_map::Entry;
+use std::collections::{HashMap, HashSet};
+use std::thread::spawn;
+use std::net::{SocketAddr};
+use mio::tcp::TcpStream;
+use serde_json::json;
+use serde::Serialize;
+
+use vm::types::{Value, QualifiedContractIdentifier, AssetIdentifier};
+use burnchains::Txid;
+use chainstate::stacks::StacksBlock;
+use chainstate::stacks::events::{StacksTransactionReceipt, StacksTransactionEvent, STXEventType, FTEventType, NFTEventType};
+use chainstate::stacks::db::StacksHeaderInfo;
+use net::StacksMessageCodec;
+
+use super::config::{EventObserverConfig, EventKeyType};
+
+#[derive(Debug)]
+struct EventObserver {
+    sock_addr: SocketAddr
+}
+
+impl EventObserver {
+
+    pub fn new(address: &str, port: u16) -> EventObserver {
+        let sock_addr = SocketAddr::new(address.parse().unwrap(), port);
+        EventObserver { sock_addr }
+    }
+
+    pub fn send(&mut self, filtered_events: Vec<&(Txid, &StacksTransactionEvent)>, chain_tip: &StacksBlock, chain_tip_info: &StacksHeaderInfo, receipts: &Vec<StacksTransactionReceipt>) {
+        // Initiate a tcp socket
+        let stream = TcpStream::connect(&self.sock_addr).unwrap();
+        
+        // Serialize events to JSON
+        let serialized_events: Vec<serde_json::Value> = filtered_events.iter().map(|(txid, event)| 
```

Comment:
Can you move the serialization code into an `impl StacksTransactionEvent`?

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1319/commits/cccd532ccb587a1e36e0e1a6690a6bfa489a8db6